### PR TITLE
23094 carry over resolution date for Horizontal & Vertical amalgamation

### PR DESCRIPTION
**Pull request by vysakh-menon-aot**
*Issue #:* /bcgov/entity#23094

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the lear license (Apache 2.0).

### CODE REVIEW

File: queue_services/entity-filer/src/entity_filer/filing_processors/filing_components/correction.py [hunk 1]
```diff
@@ -107,7 +107,6 @@ def correct_business_data(business: Business,  # pylint: disable=too-many-locals
     # update resolution, if any
     with suppress(IndexError, KeyError, TypeError):
         resolution = dpath.util.get(correction_filing, '/correction/resolution')
-        filings.update_filing_json(correction_filing_rec, resolution)
```
Comment (severinbeauvais) [inline queue_services/entity-filer/src/entity_filer/filing_processors/filing_components/correction.py:110]: What is this change?

File: queue_services/entity-filer/src/entity_filer/filing_processors/filing_components/filings.py [hunk 2]
```diff
@@ -33,14 +33,3 @@ def update_filing_court_order(filing_submission: Filing, court_order: Dict) -> O
...
     return None
-
-
-def update_filing_json(filing_submission: Filing, resolution) -> Optional[Dict]:
-    """Update the filing_json info for a Filing."""
-    if resolution:
-        if not Filing:
-            return {'error': babel('Filing required before resolution can be set.')}
-
-        filing_submission.filing_json['filing']['correction']['resolution'] = resolution
```
Comment (vysakh-menon-aot) [inline queue_services/entity-filer/src/entity_filer/filing_processors/filing_components/filings.py:44]: Not sure why we have this. We get the resolution from the same property

File: queue_services/entity-filer/src/entity_filer/filing_processors/filing_components/correction.py (hunk 1 above)
Comment (vysakh-menon-aot) [reply queue_services/entity-filer/src/entity_filer/filing_processors/filing_components/correction.py:110]: Redundant. Getting resolution from `/correction/resolution` and setting to the same property

**Review by severinbeauvais — APPROVED** [review]

File: legal-api/src/legal_api/resources/v2/business/colin_sync.py [hunk 3]
```diff
@@ -154,6 +154,9 @@ def _set_shares(primary_or_holding_business, amalgamation_filing, transaction_id
     share_classes = VersionedBusinessDetailsService.get_share_class_revision(transaction_id,
                                                                              primary_or_holding_business.id)
     amalgamation_filing['shareStructure'] = {'shareClasses': share_classes}
+    business_dates = [item.resolution_date.isoformat() for item in primary_or_holding_business.resolutions]
+    if business_dates:
+        amalgamation_filing['shareStructure']['resolutionDates'] = business_dates
```
Comment (vysakh-menon-aot) [inline legal-api/src/legal_api/resources/v2/business/colin_sync.py:159]: missed to include it in one of the ticket while estimating it

**Review by argush3 — APPROVED** [review]

**Pull request merged**